druid/src/core.rs

```diff
@@ -1002,9 +1076,33 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
             self.inner.lifecycle(&mut child_ctx, event, data, env);
         }
 
+        if let LifeCycle::DisabledChanged(_)
+        | LifeCycle::Internal(InternalLifeCycle::RouteDisabledChanged) = event
+        {
+            self.state.children_disabled_changed = false;
+            if !self.state.is_disabled() {
+                ctx.widget_state.focus_chain.extend(&self.state.focus_chain);
+            }
+
+            //Delete changes of disabled state that happened during DisabledChanged
+            self.state.is_explicitly_disabled_new = self.state.is_explicitly_disabled;
+        } else if self.state.is_explicitly_disabled != self.state.is_explicitly_disabled_new {
+            ctx.widget_state.children_disabled_changed = true;
+        }
+
         ctx.widget_state.merge_up(&mut self.state);
 
         // we need to (re)register children in case of one of the following events
+        if let LifeCycle::WidgetAdded = event {
+            // If this widget is not already present, autofocus is not set before the event,
+            // since we call register_for_focus in widget added.
+            // To be consistent with RouteWidgetAdded DisabledChanged and Route DisabledChange we
+            // insert our id at the first position.
+            if self.state.auto_focus {
+                self.state.focus_chain.insert(0, self.state.id);
+            }
+        }
+
```

Comment:
I fixed some of them and added documentation to explain the rest.

Reply:
Hmmm :/ It wasn't as easy as i thought. I think some of this complexity is necessary. 

- The ID of this widget should be the first in our focus-chain
- we should add our ID only if auto_focus is set
- auto_focus is not set until after `WidgetAdded`
- we only need to add the value if children_changed or children_disabled_changed is set, otherwise we duplicate the values.
- some_one could call `ctx.children_changed()` while routing `WidgetAdded`, which makes no sense, but therefore after the call we don't know if the value was set

The current implementation always adds the own id when the focus_chain is cleared and focus_chain is set. Only in `WidgetAdded` the id is always added after the event.
I don't know if there is a better way to satisfy the the conditions i mentioned?